### PR TITLE
DevEx #866 Utilize Sleep

### LINKED ISSUE #866: [DevEx]: Define clientside `sleep()` utility to replace use of `setTimeout()` calls for animation processing
### Improvement Summary

We should define and consume a new `sleep()` utility to streamline the use of delays in clientside code.

### Detailed Description

The client uses various delays in js to manage multi-part transitions by changing some variables in state then doing the rest after a delay using `setTimeout()`. This requires nesting the delayed code in a callback. To simplify it we could use a `sleep()` utility that returns a promise which resolves after a specified delay, allowing us to 

```
await sleep(1000); // wait 1s
doSomething();
```

instead of

```
setTimeout(() => {
doSomething()}, 1000);
```

### PR DESCRIPTION
**Pull request by Haviles04**
## Issue number

Relevant [issue number](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- Resolves #866 

## Please check the following

- [x] Do the tests still pass? (see [Run the Tests](https://github.com/cuttle-cards/cuttle/blob/main/docs/setup-development.md#run-the-tests))
- [ ] Is the code formatted properly? (see [Linting (Formatting)](https://github.com/cuttle-cards/cuttle/blob/main/docs/setup-development.md#linting-formatting))
- For New Features:
  - [ ] Have tests been added to cover any new features or fixes?
  - [ ] Has the documentation been updated accordingly?

## Please describe additional details for testing this change

-Imported sleep and replaced `setTimeout`

### CODE REVIEW

**Review by itsalaidbacklife — APPROVED** [review]
Nicely done

**Review by itsalaidbacklife — COMMENTED** [review]
Not sure what the heck happened to cause these test failures though

File: src/plugins/sockets/inGameEvents.js [hunk 1]
```diff
@@ -25,9 +25,9 @@ export async function handleInGameEvents(evData) {
   }
   // Handle GameOver
   if (evData.victory && evData.victory.gameOver) {
-    setTimeout(() => {
+    sleep(1000).then(() => {
       gameStore.setGameOver(evData.victory);
-    }, 1000);
+    });
```
Comment (Haviles04) [inline src/plugins/sockets/inGameEvents.js:30]: I tried to use async await here like I did in the other parts of the PR but for some reason, it does not work unless it's promise chained

Comment (itsalaidbacklife) [reply src/plugins/sockets/inGameEvents.js:30]: @seriouslysean any idea why tf `.then()` works but `await` doesn't? It looks to be the case that the subsequent lines fire at the appropriate time either way, but when using `await` the DOM fails to update.

**Review by itsalaidbacklife — APPROVED** [review]

File: src/plugins/sockets/inGameEvents.js (hunk 1 above)
Comment (itsalaidbacklife) [reply src/plugins/sockets/inGameEvents.js:30]: Created #923 to follow up

**Review by itsalaidbacklife — APPROVED** [review]

File: src/plugins/sockets/inGameEvents.js (hunk 1 above)
Comment (seriouslysean) [reply src/plugins/sockets/inGameEvents.js:30]: Lemme dig in to this a bit today and see if I can suss anything out.

Comment (seriouslysean) [reply src/plugins/sockets/inGameEvents.js:30]: Just to be sure, you were doing something like this?

```
// Handle GameOver
if (evData.victory?.gameOver) {
  await sleep(1000);
  gameStore.setGameOver(evData.victory);
}
```

Comment (Haviles04) [reply src/plugins/sockets/inGameEvents.js:30]: Yea

Comment (seriouslysean) [reply src/plugins/sockets/inGameEvents.js:30]: @Haviles04 any chance you can get a minimal reproduction of it not working?

Comment (Haviles04) [reply src/plugins/sockets/inGameEvents.js:30]: @seriouslysean I'm at work ATM but can do it either tonight or tomorrow

**Pull request merged**